web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx

```diff
@@ -77,6 +77,22 @@ export function SelectResource({ onSelect }: SelectResourceProps) {
 
   const [search, setSearch] = useState('');
   const [resources, setResources] = useState<ResourceSpec[]>([]);
+  const userContext = ctx.storeUser.state;
+  const { acl, authType } = userContext;
+  // sorry this looks gnarly.
+  // a user must be able to create tokens AND have access to create at least one
+  // type of resource in order to be considered eligible to "add resources"
+  // This matches how we determine "hasAccess" for each resource type but avoids
+  // having to iterate through and check for some truthy value in our resources array.
```

Comment:
Iterating through what kind of resources array we're talking about here? If it's just some static array in the code, wouldn't it be better to iterate through that so that we're sure that this check here is up to date? Otherwise adding a new kind of resource will require updating this check as well (which is not going to be easy to remember about).

Reply:
The update avoids my giant logic bubble as well, so a much greater change